lib/slow_query_logger.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'digest'
+require 'logger'
+
+# Logs slow queries to a file as reported by ActiveSupport::Notifications
+# Only queries annotated with a source will be logged.
+# For example, PreservedObject.annotate('source=MoabOnStorageService.num_preserved_objects').count
+# Adapted from https://sosedoff.com/2020/02/18/rails-log-slow-queries.html
+class SlowQueryLogger
+  def initialize(output:, threshold:)
+    @threshold = threshold
+
+    @logger = Logger.new(output || $stdout)
```

Comment:
Worth considering moving the default value here (`$stdout`) into the method definition?

```ruby
def initialize(output: $stdout, threshold:)
```

❓

Reply:
Done.